crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2289,24 +2297,33 @@ impl<'db> TypeInferenceBuilder<'db> {
                 .map(Type::IntLiteral)
                 .unwrap_or_else(|| builtins_symbol_ty(self.db, "int").to_instance(self.db)),
 
+            // Division by zero
+            // TODO: handle division by zero for floats
```

Comment:
I'd remove this TODO as it's not clear that we ever need to do this.

(Interestingly, it looks like neither mypy nor pyright consider division by zero for integers to be a type error. I think we should, since we can, but I don't think this feature alone justifies adding a float literal type.)

Reply:
We can support a diagnostic for `1.0 / 0` today without tracking float literals though, right?

Reply:
Oh true, I guess I didn't know which this todo referred to. And that reminds me, I forgot to reply to your comment about handling non-literal types. Will do that.